From c24662ab6fde4adfac35a48420f842e63d3e9831 Mon Sep 17 00:00:00 2001
From: Trent Mick <trentm@gmail.com>
Date: Thu, 8 Mar 2018 23:35:23 -0800
Subject: [PATCH] TRITON-220 TRITON-184 causes deadlock between registrar and
 config-agent in sapi which blocks startup of all (zone & GZ) config-agent's
 and many services Reviewed by: Josh Wilsdon <josh@wilsdon.ca>

---
 CHANGES.md   | 6 ++++++
 bin/agent.sh | 4 ++--
 package.json | 2 +-
 3 files changed, 9 insertions(+), 3 deletions(-)

diff --git a/CHANGES.md b/CHANGES.md
index fcb1699..ddeca4e 100644
--- a/CHANGES.md
+++ b/CHANGES.md
@@ -1,5 +1,11 @@
 # sdc-config-agent changelog
 
+## 1.6.3
+
+- [TRITON-220] Fix accidental breakage in TRITON-184 that could cause SAPI
+  zone deadlock on reboot of the sapi zone, and if the whole headnode rebooted
+  would then cause deadlock of all config-agents on that headnode.
+
 ## 1.6.2
 
 - [TOOLS-1983] Use an sdcnode built for the GZ, and bump to latest node v0.10.
diff --git a/bin/agent.sh b/bin/agent.sh
index 194d8fc..1875885 100755
--- a/bin/agent.sh
+++ b/bin/agent.sh
@@ -38,8 +38,8 @@ CONFIG_FILE=
 POLL_INTERVAL_S=120
 
 SMF_INST=${SMF_FMRI##svc:*:}
-RUN_DIR=/var/run/config-agent-$SMF_INST
-FIRST_RUN_FILE=$RUN_DIR/first-run
+RUN_DIR=/var/config-agent
+FIRST_RUN_FILE=$RUN_DIR/$SMF_INST.first-run
 
 
 # ---- support functions
diff --git a/package.json b/package.json
index 3a45773..e63bf2e 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
     "name": "config-agent",
     "description": "SmartDataCenter Config Agent",
-    "version": "1.6.2",
+    "version": "1.6.3",
     "author": "Joyent (joyent.com)",
     "dependencies": {
         "assert-plus": "0.1.5",
-- 
2.21.0

